src/common/history.c

```diff
@@ -870,66 +870,76 @@ int dt_history_copy_and_paste_on_selection(int32_t imgid, gboolean merge, GList
   return res;
 }
 
-void dt_history_compress_on_image(int32_t imgid)
+void basic_dt_history_compress_on_image(int32_t imgid)
 {
   // make sure the right history is in there:
   dt_dev_write_history(darktable.develop);
```

Comment:
this should probably be in dt_history_compress_on_image() only as this can be called from lighttable nothing to do.

Reply:
Well, maybe safe to keep it there after all. If a style is applied and the history not properly written after... Probably safer to ensure that history is properly written before dealing with it. So I withdraw my comment. Do you agree with the analysis?

Reply:
Yes i do. Would keep it